### PR TITLE
Update and reorganise use cases in DG

### PR DESCRIPTION
**Pull request by tianyue58**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -628,17 +636,99 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 1b. The user fails to enter the correct format.
+
+    * 1b1. InternSHIP shows an error message.
+
+      Use case resumes at step 1.
+
+**Use case: List upcoming deadlines**
+
+**MSS**
+
```
Comment (willyamped) [inline docs/DeveloperGuide.md:677]: Hmm this part is repeated

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -628,17 +636,99 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 1a. The list is initially empty.
 
   Use case ends.
 
-**Others**
+**Use case: Sort applications by a specific field**
+
+**MSS**
+
+1.  User requests to sort the application list by a field. The fields can be the company name, internship position, application deadline, application priority and the interview date and time.
```
Comment (willyamped) [inline docs/DeveloperGuide.md:714]: Does mentioning the fields have to do with unnecessary UI details?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (tianyue58) [reply docs/DeveloperGuide.md:677]: ohh yea I'll delete this part

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (tianyue58) [reply docs/DeveloperGuide.md:714]: okok i'll remove the details here

**Review by Nature711 — APPROVED** [review]
LGTM! Thx Tianyue :D

**Pull request merged**